Frameworks/UIKit/UILabel.mm

```diff
@@ -44,61 +44,106 @@ @implementation UILabel {
     UILineBreakMode _lineBreakMode;
     BOOL _adjustFontSize;
     float _minimumFontSize;
-    float _originalFontSize;
     int _numberOfLines;
     BOOL _isDisabled;
     BOOL _isHighlighted;
     UIBaselineAdjustment _baselineAdjustment;
     StrongId<WXCTextBlock> _textBlock;
 }
 
-- (void)adjustFontSizeToFit {
-    if (_numberOfLines == 0) {
-        [self adjustTextLayerSize];
-        return;
-    }
-
-    float curFontSize = _originalFontSize;
-    _font = [_font fontWithSize:curFontSize];
-
-    CGRect rect;
-    rect = [self bounds];
-    if (rect.size.width == 0.0f || rect.size.height == 0.0f) {
-        return;
-    }
+// Helper searches max font size for a string that can fit into given rect using given font/linkbreak config .
+// if no fit is found (e.g., to satisfy minimum font size requirement, nil is returned.
+// Otherwise, it returns the maximum font size for that can fit
+- (UIFont*)_findMaxFontSizeToFit:(CGRect)rect
```

Comment:
I added three tests in XamlCatalog, which change the minimum font size to illustrate adjustFontSize behavior is different when we can not find the fit font. Search for. 
 // number of Lines = 2, unlimited, 3 and adjustFontSize YES, TailTruncation, change minimum size to different values

---
In reply to: [98052549](https://github.com/Microsoft/WinObjC/pull/1768#discussion_r98052549) [](ancestors = 98052549)